docs/proposals/improve-addons.md

```diff
@@ -80,230 +84,176 @@ See below for a high level architecture view. Each component will be explained i
+
+### Updates to KKP API
+
+- adding an endpoint to list compatible Applications given a set of constraints (eg k8s version). This will be required to make it possible for users to select applications in the cluster wizard (see #6000)
+- adding an endpoint to install an application in a user cluster. It is important to note that it only creates the CRs which Application should be installed in which cluster, but does not do the installation of the Application itself (this task will be done by the ApplicationInstallationController).
+
+In order to do, this the following changes need to be made to the cluster and clustertemplate objects
+
+**cluster**
 
-### AddonController
+We propose a similar approach to what we currently do with the `nodeDeployment` field. Specifically, we would extend the cluster object with a new field `applications` which contains a list of all Applications as [ApplicationRefs](###ApplicationInstallation-CR), value overrides etc. We think it is necessary to (initially) store this in cluster object, so a user can set Applications directly at cluster creation. To alleviate the size issue we propose a clean-up mechanism like the `nodeDeployment` has. Specifically the `cluster` controller would go through the list of Applications and create an ApplicationInstallation for each of the items. Afterwards the controller is going to delete the entry from the list. This will reduce bloat on the cluster object.
 
-The AddonController resides in the seed cluster. Each seed cluster has its own instance of an AddonController. It has two main jobs:
+This might lead to the question: Why use ApplicationInstallation CRs at all? For this, we think having the CR is going to make routine operations (e.g. fetching, updating Applications) much easier and less resource-intensive.
 
-- selecting compatible addons from the AddonCatalogue for each user cluster
-- managing AddonInstallations for each user cluster. It is important to note that it only tracks which addon should be installed in which cluster, but does not do the installation of the addon itself (this task will be done by the AddonInstallation controller).
-- providing a query-endpoint/function that takes in a clusterConstraints and returns a list of all available addons. We need this functionality to later on make it possible for users to select addons in the cluster wizard (see #6000). We think it makes sense to include this already in the proposal, so we build an architecture that will work for future requirements
+Alternatives considered: We considered the idea of creating CRs directly on the POST call in the api. This theoretically would have the advantage that you would not need to store all AddonRefs and their values in the cluster object. We decided to not go with this for the following reasons:
```

Comment:
For initial addons/applications (ones created with the cluster; from wizard) I'd say it would be the best to keep them as part of the cluster object (perhaps in the annotation like initial node deployment) and then create addons/application CR? For the existing clusters I'd use CRs only.

cc @kubermatic/sig-ui @kubermatic/sig-ui for more opinions on this

Reply:
@mfranczy changed to ApplicationRef